changes/2020-06-17_encrypt-max-plaintext-length-input/change.md

```diff
@@ -0,0 +1,182 @@
+
+This input already exists in the specification as [plaintext length](../../client-apis/encrypt.md#plaintext-length)
+however its effect on the Encrypt operation's behavior is unspecified.
+
+This change renames this input in the specification
+to be a more accurate reflection of its behavior,
+and specifies the correct behavior during the Encrypt operation.
+
+This change also forbids exposing any API which only performs Encrypt on known-length plaintexts
+from allowing users to specify this input.
+
+## Out of Scope
+
+- How different implementations may allow a user to specify this input for the Encrypt operation
+  is out of scope.
+- Significantly changing the shape of any of our Encrypt API implementations is out of scope.
+
+## Motivation
+
+All implementations allow users to specify a plaintext length when calling Encrypt with a
+unknown-length plaintext.
+However, its behavior in the Encrypt operation is not clearly specified.
+
+Additionally, some implementations also allow a plaintext length to be specified on Encrypt with a
+known-length plaintext.
+The behavior of Encrypt in this case is not consistent between implementations.
+
+The purpose of this change is to document the correct behavior of this input and
+rename this input to clarify its intent.
+
+This change renames this input within the spec because
+the intention behind the value of this input is that
+it is a bound on the plaintext length,
+not the exact value.
+This is what the Encrypt operation enforces when this value is set,
+and this is how CMMs should interpret this value in GetEncryptionMaterials if it is set.
+Implementations already treat this input with this intent in mind,
+so we should describe it in the specification as such.
+
+This change specifies that the Encrypt operation MUST pass the real length of the plaintext
+to the CMM if known instead of a user supplied bound because the former is more accurate.
+There is no case where a user would benefit from passing
+a less accurate `max plaintext length` to the CMM.
+If there were such a case,
+then such a CMM is not correctly using the intent behind `maxPlaintextLength`.
+It is better to restrict flexibilty here so that users have less opportunity
+to depend on behaviors from CMMs that are misusing this value.
+
+## Drawbacks
+
+This change does not prescribe the name of this input in implementations,
+and thus there will not be a standardized name or way of specifying this input across implementations.
+This is because this control already exists in implementations under different names,
+and it is not worth breaking all customers to rename this input
+without significantly changing this input to benefit the customer.
+
+## Security Implications
+
+This change SHOULD NOT have any security implications.
+
+## Operational Implications
+
+This change will break any JS user that is currently using Encrypt with a known-length plaintext
+and inputting a plaintextLength less than that length.
```

Comment:
No, the 2 JS implementations do not let this happen.
The browser does not accept the option at all because **all** operations are one-shot.
https://github.com/aws/aws-encryption-sdk-javascript/blob/master/modules/encrypt-browser/src/encrypt.ts#L40
The Node implementation will clobber any value provided in favor of the correct value. 
https://github.com/aws/aws-encryption-sdk-javascript/blob/master/modules/encrypt-node/src/encrypt.ts#L30

Reply:
I will update to specify "Node JS". The Node JS implementation violates the spec I've outlined here. Instead of clobbering in the "less than" case, it MUST fail. Therefore, anyone who was previously depending on the implementation just using the correct value in the "less than" case will break.

Reply:
Hum, if what you want is to "not let it be an option" then how would that work, if not this way?